e2e_tests/tests/experiment/noop.py

```diff
@@ -0,0 +1,173 @@
+    def __init__(self, msg: str, level: int = logging.INFO) -> None:
+        # Transfer message as base64 to allow weird characters, and also so that when searching logs
+        # you don't see the msg string anywhere in "metadata", like expconf.
+        self.base64 = base64.b64encode(msg.encode("utf8")).decode("utf8")
+        self.level = level
+
+    def to_dict(self) -> Dict[str, Any]:
+        return {"action": "log", "base64": self.base64, "level": self.level}
+
+
+class CompleteSearcherOperation:
+    def __init__(self, metric: float) -> None:
+        self.metric = metric
+
+    def to_dict(self) -> Dict[str, Any]:
+        return {"action": "complete_searcher_operation", "metric": self.metric}
+
+
+Action = Union[Exit, Sleep, Report, Checkpoint, Log, CompleteSearcherOperation]
+
+
+def merge_config(old: Any, new: Any) -> Any:
+    if isinstance(old, dict):
+        out = dict(old)
+        for key, new_value in new.items():
+            if key in out:
+                out[key] = merge_config(out[key], new_value)
+            else:
+                out[key] = new_value
+        return out
+    return new
+
+
+def generate_config(
+    *actions: Action,
+    config: Optional[Dict[str, Any]] = None,
+) -> Dict[str, Any]:
+    base_config = {
+        "name": "noop",
+        "entrypoint": "python3 train.py",
+        "hyperparameters": {
+            "actions": {str(i): a.to_dict() for i, a in enumerate(actions, start=1)},
+        },
+        "max_restarts": 0,
+        "searcher": {
+            "name": "single",
+            "metric": "x",
+            "max_length": 1,  # XXX remove this
+        },
+    }
+
+    if config is None:
+        return base_config
+    else:
+        return merge_config(base_config, config)  # type: ignore
+
+
+def create_experiment(
+    sess: api.Session,
+    *actions: Action,
+    config: Optional[Dict[str, Any]] = None,
+    project_id: Optional[int] = None,
+    includes: Optional[Iterable[Union[str, pathlib.Path]]] = None,
+    template: Optional[str] = None,
+) -> client.Experiment:
+    config = generate_config(*actions, config=config)
+    model_dir = conf.fixtures_path("noop")
+    return client.Determined._from_session(sess).create_experiment(
+        config, model_dir, project_id=project_id, includes=includes, template=template
+    )
+
+
+def cli_config_overrides(*actions: Action) -> List[str]:
+    """
+    Return a list of `--config=...` to override the actions hyperparameter for the CLI.
+
+    Note that you must provide the full list of actions (which makes your test more readable anyway)
+    and also note that you can't delete actions through this mechanism, because that isn't supported
+    by our --config override mechanism.
+    """
+
+    return [
+        f"--config=hyperparameters.actions.{action_id}={json.dumps(action.to_dict())}"
+        for action_id, action in enumerate(actions, start=1)
+    ]
+
+
+def traininglike_steps(count: int, scale: float = 1.0) -> List[Action]:
```

Comment:
count seems kinda weird to me. also might be useful to have max length. also just "scale" is hard to understand. 

what about max_length / step_period / metric_scale_factor (or just metric_scale?)

Reply:
I can do `max_length` and `metric_scale`, I like that.

Reply:
done